homeassistant/components/point/__init__.py

```diff
@@ -70,62 +82,102 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
     conf = config[DOMAIN]
 
-    config_flow.register_flow_implementation(
-        hass, DOMAIN, conf[CONF_CLIENT_ID], conf[CONF_CLIENT_SECRET]
+    await async_import_client_credential(
+        hass,
+        DOMAIN,
+        ClientCredential(
+            conf[CONF_CLIENT_ID],
+            conf[CONF_CLIENT_SECRET],
+        ),
     )
 
-    hass.async_create_task(
-        hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}
-        )
+    async_create_issue(
+        hass,
+        HOMEASSISTANT_DOMAIN,
+        f"deprecated_yaml_{DOMAIN}",
+        breaks_in_ha_version="2024.8.0",
+        is_fixable=False,
+        issue_domain=DOMAIN,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+        translation_placeholders={
+            "domain": DOMAIN,
+            "integration_title": "Point",
+        },
+    )
+
+    await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data=conf,
     )
 
     return True
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up Point from a config entry."""
+async def async_setup_entry(hass: HomeAssistant, entry: PointConfigEntry) -> bool:
+    """Set up Minut Point from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
 
-    async def token_saver(token, **kwargs):
-        _LOGGER.debug("Saving updated token %s", token)
+    if "auth_implementation" not in entry.data:
+        # Config entry is imported from old implementation without native oauth2.
```

Comment:
Can we do this migration directly in the config flow import step?

Reply:
Perhaps, but my tests suggests that I need to do a reauth anyway so I guess this could be removed. Or we need to access the `entry.data` from the `import` step (which I don't know how to do).

I believe that I am the only one still using this integration due to the time it's taken to get this fix in motion, so perhaps it is not super important as this part of code will be removed eventually.

Reply:
What is the current entry data like? Should we import the oauth credentials and ask for a reauth and have HA set the data as we want?

Reply:
Yes, we would like to import the oauth credentials and ask for a reauth and have HA set the data as we want. This is actually what's happening on my tests, but I guess it can/should be made more elegantly.

As previous discussion, there are so few (23) active installations running so I think we could skip all migration and let the brave 22 others than me manually reinstall.

Reply:
> The migration code for this is quite slim, it's just importing and asking for a reauth

So we should keep it as it is now?

Reply:
So wait, this integration has both YAML and config entries? Waht is currently stored? We should import the client id and secret, trigger a reauth, then the config entry should be like normal.

Reply:
Can you point me at some code how to do this?

We currently import the client_id and client_secret from yaml and trigger a reauth.